src/libraries/System.Private.CoreLib/src/System/ValueTuple.cs

```diff
@@ -337,12 +337,12 @@ int IComparable.CompareTo(object? other)
         /// instance is equal to <paramref name="other"/>, and greater than zero if this instance is greater
         /// than <paramref name="other"/>.
         /// </returns>
-        public int CompareTo(ValueTuple<T1> other)
+        public readonly int CompareTo(ValueTuple<T1> other)
         {
             return Comparer<T1>.Default.Compare(Item1, other.Item1);
         }
 
-        int IStructuralComparable.CompareTo(object? other, IComparer comparer)
+        readonly int IStructuralComparable.CompareTo(object? other, IComparer comparer)
```

Comment:
Putting `readonly` on explicit interface implementations isn't going to buy you anything. These can only be accessed via interface references and those don't have defensive copy semantics. 

I'm actually a bit surprised we allow this syntax.